homeassistant/helpers/entity.py

```diff
@@ -325,6 +325,22 @@ def has_entity_name(self) -> bool:
             return self.entity_description.has_entity_name
         return False
 
+    def _device_class_name(self) -> str | None:
+        """Return a translated name of the entity based on its device class."""
+        assert self.platform
+        if not self.has_entity_name:
+            return None
+        device_class_key = self.device_class or "_"
+        name_translation_key = (
+            f"component.{self.platform.domain}.entity_component."
+            f"{device_class_key}.name"
+        )
+        return self.platform.component_translations.get(name_translation_key)
+
+    def _default_to_device_class_name(self) -> bool:
+        """Return True if an unnamed entity should be named by its device class."""
+        return False
+
     @property
     def name(self) -> str | None:
         """Return the name of the entity."""
```

Comment:
We don't change the default of `_attr_name` to `UNDEFINED`. Shouldn't we do that?

Reply:
`Entity._attr_name` has no default value since the attribute is not defined when an `Entity` object is created, it just has a type annotation. I was intending to change the default return value of the `name` property from `None` to the `UNDEFINED` sentinel in a separate PR.